piqueserver/game_modes/freeforall.py

```diff
@@ -52,16 +74,37 @@ def on_flag_spawn(self, x, y, z, flag, entity_id):
                 return HIDE_POS
             return protocol.on_flag_spawn(self, x, y, z, flag, entity_id)
 
+        def broadcast_contained(self, contained, unsequenced=False,
+                                sender=None, team=None, save=False, rule=None):
+            if ISOLATE_PLAYER.get():
+                if contained.id == CreatePlayer.id:
+                    if contained.team != -1:
+                        player = self.players[contained.player_id]
+                        contained.team = self.blue_team.id
+
+                        player.send_contained(contained)
+                        contained.team = self.green_team.id
+                        sender = player
+
+            return protocol.broadcast_contained(self, contained, unsequenced,
+                                                sender, team, save, rule)
+
     class FreeForAllConnection(connection):
         score_hack = False
 
+        def on_team_join(self, team):
+            if team.spectator or not ISOLATE_PLAYER.get():
+                return team
+
+            return self.protocol.green_team
```

Comment:
I think there's a `.team_1` member variable to avoid the confusion with team names?